### DISCUSSION TITLE
Websockets in NGINX not working with server with internet proxy: Error: No pong received in 3 seconds

### DISCUSSION THREAD

**Discussion by JithinThomasDataSemantics** [miguelgrinberg/python-socketio#762] [Q&A]
I was trying to host flask application in NGINX which uses websockets. It is working fine with the servers which do not use any proxy servers. When it is hosted in a server that passes requests to proxy servers, client does not receive any message sent via websocket. Initially none of the external API calls were working which started working when I added environ variable http_proxy and https_proxy for the service. But the socket is still not working.

Got error: "no pong received in 3 seconds" in the server when trying to connect to websocket

```
'msec $msec request_time $request_time';
upstream socket_nodes {
    ip_hash;
    server 127.0.0.1:4000;
    server 127.0.0.1:4001;
    server 127.0.0.1:4002;
}
server {
        listen 80;
        listen [::]:80;
        access_log /var/log/nginx/access.log upstreamlog;
        add_header Strict-Transport-Security max-age=15768000;
        location /static/* {
            alias /file_path;
        }
        location / {
            include uwsgi_params;
            proxy_pass      http://socket_nodes;

            proxy_redirect  off;
            proxy_set_header        Host    $host;
            proxy_set_header        X-Real-IP       $remote_addr;
            proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header        X-Forwarded-Proto       $scheme;
            add_header      Front-End-Https on;
            proxy_buffer_size          16k;
            proxy_busy_buffers_size    16k;
        }
        location /socket.io {
            proxy_pass http://socket_nodes/socket.io;
            proxy_http_version 1.1;
            proxy_redirect off;
            proxy_buffering off;
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "upgrade";
        }
}
```

**Comment by miguelgrinberg** [comment]
This is an issue between nginx and uWSGI, and has nothing to do with Flask-SocketIO. I don't really use uWSGI that much, so I can't really help. Have you searched for the error that you get? A lot of people have this problem.

**Reply by JithinThomasDataSemantics** [reply]
I enabled the logger  logger=False, engineio_logger=False , and the error in backend was 
**No pong received in 3 seconds**
in the UI there is no error shown the socket connection 
![image](https://user-images.githubusercontent.com/47312179/128854289-fd66c546-6a2c-423a-85bc-58437893ae3f.png)

**Reply by miguelgrinberg** [reply]
As I said, the error "No pong received in 3 seconds" does not come from Flask-SocketIO. It comes from uWSGI, you have to look at the interaction between uWSGI and nginx to solve your problem.

**Reply by JithinThomasDataSemantics** [reply]
Sure @miguelgrinberg  Thanks for your response.. will update if we could solve this
